stablehlo/dialect/Base.cpp

```diff
@@ -33,6 +33,24 @@ limitations under the License.
 namespace mlir {
 namespace hlo {
 
+bool allUniformQuantizedType(TypeRange typeRange) {
```

Comment:
For non-shaped types it would crash.

How about we resuse the functions from [TypeInference.cpp](https://github.com/openxla/stablehlo/blob/c3f456500f0f2e96fdb4a98fde4fbe48b4d624b8/stablehlo/dialect/TypeInference.cpp#L72C1-L84C2) ( with small mofication) and put them in Base.h 

```
template <typename T>
bool allQuantized(ArrayRef<Type> typeRange) {
  return llvm::all_of(typeRange, [&](Type type) {
    return getElementTypeOrSelf(type).isa<T>();
  });
}

template <typename T>
bool noneQuantized(ArrayRef<Type> typeRange) {
  return llvm::all_of(typeRange, [&](Type type) {
    return !getElementTypeOrSelf(type).isa<T>();
  });
}
```